include/arch/arc/v2/linker.ld

```diff
@@ -111,6 +118,18 @@ SECTIONS {
 #include <snippets-rodata.ld>
 
 #include <linker/kobject-rom.ld>
+
+#if defined(CONFIG_CPLUSPLUS) && !defined(CONFIG_CPP_STATIC_INIT_GNU)
+		. = ALIGN(4);
+		_fctors = .;
+		KEEP(*(.ctors*))
+		_ectors = .;
+
+		. = ALIGN(4);
+		_feh_frame = .;
+		KEEP(*(.eh_frame*))
```

Comment:
this looks like exception handling, not constructors: is it required? Do we now support exception handlers as well for MWDT?

Reply:
Yep, that's a bit tricky.
We don't need `.eh_frame` for constructors initialization - it's used for exception handling. However this section cames from the additional library we link so we need to specify memory region for it (otherwise we'll break the compilation).

So there is two possible approaches:
 1. Just drop `eh_frame` (i.e add it to `/DISCARD/` sections) as C++ exceptions are still WIP for MWDT toolchain in Zephyr. In that case we can bring it back when we'll fully add C++ exceptions support.
 2. [_current state_] Put in in proper place and add res of the things when we'll add  C++ exceptions support for Zephyr in MWDT.
 
 I'm OK wit both approaches - so I can switch to the first one if you want.

Reply:
current approach is fine - maybe make it clear in the 'TODO' that exceptions still need to be done.